components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/ActionListenerImpl.java

```diff
@@ -46,34 +45,43 @@
 import static org.wso2.carbon.identity.remotefetch.core.util.RemoteFetchConfigurationUtils.generateUniqueID;
 
 /**
- * ActionListener that polls repository with frequency for changes to be deployed.
+ * Action Listener parent class to create, manage deploy revisions.
  */
-public class PollingActionListener implements ActionListener {
+public class ActionListenerImpl implements ActionListener {
 
-    private static final Log log = LogFactory.getLog(PollingActionListener.class);
+    private static final Log log = LogFactory.getLog(ActionListenerImpl.class);
 
     private RepositoryManager repo;
-    private Integer frequency;
-    private Date lastIteration;
+    protected Date lastIteration;
     private ConfigDeployer configDeployer;
     private DeploymentRevisionDAO deploymentRevisionDAO;
     private Map<String, DeploymentRevision> deploymentRevisionMapNotResolved = new HashMap<>();
     private Map<String, DeploymentRevision> deploymentRevisionMap = new HashMap<>();
     private String remoteFetchConfigurationId;
     private int tenantId;
 
-    public PollingActionListener(RepositoryManager repo, ConfigDeployer configDeployer,
-                                 int frequency, String remoteFetchConfigurationId, int tenantId) {
+    public ActionListenerImpl(RepositoryManager repo, ConfigDeployer configDeployer,
+                              String remoteFetchConfigurationId, int tenantId) {
 
         this.repo = repo;
         this.configDeployer = configDeployer;
-        this.frequency = frequency;
         this.remoteFetchConfigurationId = remoteFetchConfigurationId;
-        this.deploymentRevisionDAO = new DeploymentRevisionDAOImpl();
         this.tenantId = tenantId;
+        this.deploymentRevisionDAO = new DeploymentRevisionDAOImpl();
         this.seedRevisions();
     }
 
+    @Override
+    public void iteration() {
```

Comment:
Looks like this method is not an iteration at all. This is better rem=named as "execute" or "process"